ssl/statem/statem_srvr.c

```diff
@@ -3229,6 +3229,7 @@ static int tls_process_cke_gost(SSL_CONNECTION *s, PACKET *pkt)
         return 0;
     }
     if (EVP_PKEY_decrypt_init(pkey_ctx) <= 0) {
+        EVP_PKEY_CTX_free(pkey_ctx);
         SSLfatal(s, SSL_AD_INTERNAL_ERROR, ERR_R_INTERNAL_ERROR);
         return 0;
```

Comment:
A fix more consistent with the rest of this function might be to instead replace this `return 0` with a `goto err`